src/NovelAudioService.cpp

```diff
@@ -329,6 +318,31 @@ std::string NovelAudioService::findByChannelMap(int channel) {
   return nullptr;
 }
 
+void NovelAudioService::logIfSDLFailure(int (*function)(Uint32), Uint32 sdl_flag, std::string errorMessage) {
+  if (function(sdl_flag) < NovelUtilities::SDL_SUCCESS)
+  {
+    _logger.log(errorMessage, LogLevel::ERR);
+    throw std::runtime_error("Audio error occurred! Unable to continue.");
+  }
+}
+
+void NovelAudioService::logIfMixerFailure(int (*function)(int), int mixerFlag, std::string errorMessage) {
+  if (function(mixerFlag) < NovelUtilities::SDL_SUCCESS)
+  {
+    _logger.log(errorMessage, LogLevel::ERR);
+    throw std::runtime_error("Audio error occurred! Unable to continue.");
+  }
+}
+
+void NovelAudioService::logIfMixerFailure(int (*function)(int, Uint16, int, int), int freq, Uint16 mixerFormat, int channels, int sampleSize, std::string errorMessage) {
+  if (function(freq, mixerFormat, channels, sampleSize) < NovelUtilities::SDL_SUCCESS)
+  {
+    _logger.log(errorMessage, LogLevel::ERR);
+    throw std::runtime_error("Audio error occurred! Unable to continue.");
+  }
+}
+
+
```

Comment:
This looks cringey to me - @tannergooding should I attempt to use `...` here and make the function have variable params?
I was originally going to, but it seemed like I would be including a lot of extra stuff for only two functions.

Reply:
Looks like it would really be a part of #71 - closing for now